pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,201 @@
+
+    class Py37Str(str):
+        # Allow Python 3.6 compatibility
+        def isascii(self: str) -> bool:
+            return all("\u0000" <= x <= "\u007F" for x in self)
+
+
+NON_ASCII_HELP = (
+    "Used when the name contains at least one non-ASCII unicode character. "
+    "See https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers"
+    " for a background why this could be bad. \n"
+    "If your programming guideline defines that you are programming in "
+    "English, then there should be no need for non ASCII characters in "
+    "Python Names. Everybody else can simply disable this check."
+)
+
+
+class NonAsciiNameChecker(base_checker.BaseChecker):
+    """A strict name checker only allowing ASCII
+
+    Note: This check only checks Names, so it ignores the content of
+          docstrings and comments!
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    priority = -1
+
+    msgs = {
+        "C2401": (
+            '%s name "%s" contains a non-ASCII character, consider renaming it.',
+            "non-ascii-name",
+            NON_ASCII_HELP,
+            {"old_names": [("C0144", "old-non-ascii-name")]},
+        ),
+        # First %s will always be "file"
+        "W2402": (
+            (
+                '%s name "%s" contains a non-ASCII character. PEP 3131 only allows '
+                "non-ascii identifiers, not file names."
+            ),
+            "non-ascii-file-name",
+            (
+                # Some = PyCharm at the time of writing didn't display the non_ascii_name_loł
+                # files and had big troubles with git.
+                # Probably only a bug shows the problem quite good.
+                # That's also why this is a warning and not only a convention!
+                "Some editors don't support non-ASCII file names properly. "
+                "Even though Python supports UTF-8 files since Python 3.5 this isn't "
+                "recommended for interoperability. Further reading:\n"
+                "- https://www.python.org/dev/peps/pep-0489/#export-hook-name\n"
+                "- https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers\n"
+                "- https://bugs.python.org/issue20485\n"
+            ),
+        ),
+        # First %s will always be "module"
+        "C2403": (
+            '%s name "%s" contains a non-ASCII character, use an ASCII-only alias for import.',
+            "non-ascii-module-import",
+            NON_ASCII_HELP,
+        ),
+    }
+
+    name = "NonASCII-Checker"
+
+    def _check_name(
+        self, node_type: str, name: Optional[str], node: nodes.NodeNG
+    ) -> None:
+        """Check whether a name is using non-ASCII characters."""
+
+        if name is None:
+            # For some nodes i.e. *kwargs from a dict, the name will be empty
+            return
+
+        if not (Py37Str(name).isascii()):
+            type_label = constants.HUMAN_READABLE_TYPES.get(node_type, node_type)
```

Comment:
```suggestion
            type_label = constants.HUMAN_READABLE_TYPES[node_type]
```

If we're missing any this will be a good check to tell us we need to update `constants.HUMAN_READABLE_TYPES` 😄